File: core/src/nl/t64/game/rpg/components/door/Door.java
package nl.t64.game.rpg.components.door;

import lombok.Getter;
import lombok.NoArgsConstructor;
import nl.t64.game.rpg.audio.AudioEvent;

import java.util.Optional;


@Getter
@NoArgsConstructor
public class Door {

    DoorType type;
    String spriteId;
    String keyId;
    AudioEvent audio;
    float width;
    float height;
    boolean isLocked;
    boolean isClosed;

    Door(String type, String spriteId) {
        this(type, spriteId, null);
    }

    Door(String type, String spriteId, String keyId) {
        this.type = DoorType.valueOf(type);
        this.spriteId = spriteId;
        this.keyId = keyId;
    }

    public Optional<String> getKeyId() {
        return Optional.of(keyId);
    }

    public boolean isLocked() {
        return isLocked;
    }

    public void unlock() {
        isLocked = false;
    }

    public boolean isClosed() {
        return isClosed;
    }

    public void open() {
        isClosed = false;
    }

    public void close() {
        isClosed = true;
    }

}

File: core/src/nl/t64/game/rpg/components/door/Door.kt
package nl.t64.game.rpg.components.door

import nl.t64.game.rpg.audio.AudioEvent


class Door(
    val type: DoorType = DoorType.SMALL,    // this will become replaced by the correct json value.
    val spriteId: String = "",
    val keyId: String? = null
) {

    lateinit var audio: AudioEvent
    var width: Float = 0f
    var height: Float = 0f
    var isLocked = false
    var isClosed = false

    fun unlock() {
        isLocked = false
    }

    fun open() {
        isClosed = false
    }

    fun close() {
        isClosed = true
    }

}
